### PR TITLE
Remove Decoding of Scheme from Action Invocations

### PR DESCRIPTION
**Pull request by dubeejw**
- No longer decode action invocation response into structs

Closes https://github.com/openwhisk/openwhisk/issues/1489

### CODE REVIEW

**Comment by dubeejw** [comment]
PG3 255

File: tests/src/system/basic/WskBasicTests.scala [hunk 1]
```diff
@@ -477,15 +477,15 @@ class WskBasicTests
                 activation =>
                     activation.response.result shouldBe Some(dynamicParams.toJson)
                     activation.duration shouldBe 0L // shouldn't exist but CLI generates it
-                    activation.end shouldBe Instant.EPOCH.toEpochMilli // shouldn't exist but CLI generates it
+                    activation.end shouldBe Instant.ofEpochMilli(0L) // shouldn't exist but CLI generates it
```
Comment (rabbah) [inline tests/src/system/basic/WskBasicTests.scala]: why dos the comment say shouldn't exist but CLI generates it - should the CLI stop generating it when this schema stripping is done?

Comment (dubeejw) [inline tests/src/system/basic/WskBasicTests.scala]: Yes, the CLI will stop automatically producing these fields when the schema is removed. Removal of the schema decoding will have to be done for each command and subcommand. For instance, this PR currently removes the decoding for action invocations only. In order to remove mentioned generated fields, trigger fire would have to be updated.

Comment (markusthoemmes) [inline tests/src/system/basic/WskBasicTests.scala]: Also: Shouldn't these two lines produce an identical result?

File: tests/src/common/WskTestHelpers.scala [hunk 2]
```diff
@@ -141,6 +141,18 @@ trait WskTestHelpers extends Matchers {
     }
 
     object CliActivation extends DefaultJsonProtocol {
+        private implicit val instantSerdes = new RootJsonFormat[Instant] {
+            def write(t: Instant) = t.toEpochMilli.toJson
+
+            def read(value: JsValue) = Try {
+                value match {
+                    case JsString(t) => Instant.parse(t)
```
Comment (markusthoemmes) [inline tests/src/common/WskTestHelpers.scala]: While this is nice to have, our code should never return a `JsString` here, so we should drop the serdes there?

File: tests/src/system/basic/WskBasicTests.scala (hunk 1 above)
Comment (dubeejw) [inline tests/src/system/basic/WskBasicTests.scala]: Instant.EPOCH.toEpochMilli == 0
Instant.ofEpochMilli(0L) == 1970-01-01T00:00:00Z

Comment (markusthoemmes) [inline tests/src/system/basic/WskBasicTests.scala]: Okay, but `Instant.EPOCH` should be equal to `ofEpochMilli(0L)`?

Comment (dubeejw) [inline tests/src/system/basic/WskBasicTests.scala]: That is correct! I'll use `Instant.EPOCH` instead.

**Comment by dubeejw** [comment]
PG1 1165.

**Comment by dubeejw** [comment]
@rabbah, every sub-command will have to have the same updates that `action invoke` have here. I'd suggest to merge each updated sub-command individual as it will take some time to update all of them.

File: tests/src/system/basic/WskBasicTests.scala [hunk 3]
```diff
@@ -487,16 +487,16 @@ class WskBasicTests
             withActivation(wsk.activation, run) {
                 activation =>
                     activation.response.result shouldBe Some(dynamicParams.toJson)
-                    activation.duration shouldBe 0L // shouldn't exist but CLI generates it
-                    activation.end shouldBe Instant.EPOCH.toEpochMilli // shouldn't exist but CLI generates it
+                    activation.duration shouldBe 0L         // shouldn't exist but CLI generates it
+                    activation.end shouldBe Instant.EPOCH   // shouldn't exist but CLI generates it
```
Comment (rabbah) [inline tests/src/system/basic/WskBasicTests.scala]: did i ask this already? the comment says "shouldn't exist" but CLI generates it - is this not being removed/addressed by this pr?

Comment (rabbah) [inline tests/src/system/basic/WskBasicTests.scala]: i did ask already and ou answered.

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/commit/a600110aa1511f08871618f45745e5b5c0cdcea4.

**Pull request closed without merging**